### PR TITLE
Bump Kubernetes to v1.24.0 and fix AWSMachinePool minsize

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:
Bumping cluster-api version to v1.1.5 and Kubernetes version we use in e2e tests to v1.24.0.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3242
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3600

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by sedefsavas** [comment]
@sbueringer following up on the [slack thread](https://kubernetes.slack.com/archives/C8TSNPY4T/p1652367225328959?thread_ts=1652363871.078409&cid=C8TSNPY4T):
There is no K8s library bump in this PR, still seeing the same issue of:
`
../../../pkg/mod/sigs.k8s.io/cluster-api@v1.1.3/internal/topology/variables/clusterclass_variable_validation.go:139:100: not enough arguments in call to structuraldefaulting.ValidateDefaults
`

File: go.mod [hunk 1]
```diff
@@ -26,20 +26,20 @@ require (
 	github.com/sergi/go-diff v1.2.0
 	github.com/spf13/cobra v1.4.0
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/crypto v0.0.0-20210921155107-089bfa567519
+	golang.org/x/crypto v0.0.0-20220214200702-86341886e292
 	golang.org/x/text v0.3.7
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.23.5
+	k8s.io/api v0.24.0
```
Comment (sbueringer) [inline go.mod:32]: I think those k8s.io/* versions all have to stay the same

Comment (sedefsavas) [reply go.mod]: Thanks! finally reverted those properly.

File: test/e2e/data/e2e_conf.yaml [hunk 2]
```diff
@@ -252,21 +252,21 @@ variables:
   # allowing the same e2e config file to be re-used in different Prow jobs e.g. each one with a K8s version permutation.
   # The following Kubernetes versions should be the latest versions with already published kindest/node images.
   # This avoids building node images in the default case which improves the test duration significantly.
-  KUBERNETES_VERSION_MANAGEMENT: "v1.23.3"
-  KUBERNETES_VERSION: "v1.23.3"
-  KUBERNETES_VERSION_UPGRADE_TO: "v1.23.3"
-  KUBERNETES_VERSION_UPGRADE_FROM: "v1.22.4"
+  KUBERNETES_VERSION_MANAGEMENT: "v1.24.0"
```
Comment (sbueringer) [inline test/e2e/data/e2e_conf.yaml:255]: I think we have to keep this version on v1.23.3 for now.

The problem is that only kind v0.13 is compatible with kindest/node:v1.24.0+ and we didn't bump the kind version yet.

(cc @fabriziopandini)

Comment (sedefsavas) [reply test/e2e/data/e2e_conf.yaml]: ACK, done.

Comment (sbueringer) [reply test/e2e/data/e2e_conf.yaml]: With the latest v1.1.x release we should now be able to use v1.24 as mgmt cluster

**Comment by Ankitasw** [comment]
@sedefsavas any leads on this PR?

**Comment by sedefsavas** [comment]
/hold
will also address https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3600 in this PR.

**Comment by sedefsavas** [comment]
All tests passed except MachinePool: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/3468/pull-cluster-api-provider-aws-e2e/1547683508405669888

`E0714 19:28:49.478383       1 awsmachinepool_controller.go:254]  "msg"="error updating AWSMachinePool" "error"="unable to update ASG: failed to update ASG \"machine-pool-nncde0-mp-0\": ValidationError: Desired capacity:0 must be between the specified min size:1 and max size:4\n\tstatus code: 400, request id: 74978a6a-e440-44cb-aef2-b95e51c49e7b" `

which will be solved by
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3253

@mweibel feel free to add your commit on top of this PR.

**Comment by mweibel** [comment]
done @sedefsavas. Thanks for the help :+1: 
I'm investigating why the MP test failed in my branch.

**Comment by sedefsavas** [comment]
Reason of the machinepool failure was e2e-config was using an old nightly image as the latest of 1.1.x capi version. 
After all the e2e tests pass, this is good to go.

**Comment by mweibel** [comment]
ah that makes sense! I did wonder whether this needs updating but wasn't sure. 

Thanks for fixing!

**Comment by sedefsavas** [comment]
/assign @richardcase 

EKS tests passed in the previous run.

**Comment by richardcase** [comment]
This all looks ok to me

/lgtm

**Comment by sedefsavas** [comment]
/approve

Ups there is a pull-cluster-api-provider-aws-build-release-1-5  job running, there must be a bug in the release-1.5 jobs.

**Pull request merged**